### ISSUE TITLE
Little gap between words on the alignment plot

### ISSUE THREAD

**Issue by dnnnew** [NVIDIA/tacotron2#135]
Hi, 
My alignment plot has a little gap between words. 

<img width="1111" alt="alignment-63k" src="https://user-images.githubusercontent.com/45612997/52099632-f1d24e00-260e-11e9-9023-8c547ebcbdb1.png">

batch size - 64
filter_length=2048,
hop_length=275,
win_length=1100,

And it sounds bit like reading separated words.
Is there anyone who had same issue before. 

Thanks.

**Comment by rafaelvalle** [comment]
The alignment looks promising. What dataset is this? Are there no gaps in the dataset itself?

**Comment by dnnnew** [comment]
Dataset is Mongolian, female, 15.3k samples and 35 hours. 
Alignment after 200k iters: 
<img width="676" alt="screen shot 2019-02-10 at 2 50 02 pm" src="https://user-images.githubusercontent.com/45612997/52530822-42038b80-2d46-11e9-9598-500ee4599639.png">

There are no gaps in the dataset. 

Here is Alignment of 250k iters https://github.com/Rayhane-mamah/Tacotron-2 same dataset (different sentence). 
![step-250000-align](https://user-images.githubusercontent.com/45612997/52530838-94dd4300-2d46-11e9-9afb-d6519d373f6a.png)

Sample wav:
[12703_020.wav.zip](https://github.com/NVIDIA/tacotron2/files/2848608/12703_020.wav.zip)

<img width="1369" alt="screen shot 2019-02-10 at 3 18 51 pm" src="https://user-images.githubusercontent.com/45612997/52530864-3795c180-2d47-11e9-84ca-5b67a7f0d699.png">

**Comment by dnnnew** [comment]
It isn't solved yet. I am testing another hparams, changing dataset's parts.

**Comment by rafaelvalle** [comment]
Let us know. I'll keep it open until we fix it.

**Comment by Engineering-Course** [comment]
Hi @dnnnew,
Do you find out what causes the gaps?
I also meet the same problem on my dataset.
![image](https://user-images.githubusercontent.com/7713394/53150030-77339780-35ea-11e9-941a-7eaa5022185e.png)

**Comment by dnnnew** [comment]
@Hi, Engineering-Course.

Not yet.
I guessed it might be depend on my dataset.

My dataset's wav file has multiple sentence or incomplete sentence. So I am working on my dataset (realignment wav and text) now then will train again. I'll report result.

**Comment by dnnnew** [comment]
@Engineering-Course 
please let me know If you find out what causes the gaps.

**Comment by Engineering-Course** [comment]
However, the wav files on my dataset have the complete sentence and no silence.
Does sampling_rate matter? @rafaelvalle 
The sampling rate of my wav is 48000. Then I change the sampling rate in hparams.py from 22050 to 48000.

**Comment by Engineering-Course** [comment]
BTW, What's your tool to visualize the wave like this? @dnnnew 
![image](https://user-images.githubusercontent.com/7713394/53152062-94b73000-35ef-11e9-975b-3c8f226a34fa.png)

**Comment by dnnnew** [comment]
Adobe Audition @Engineering-Course.

**Comment by Engineering-Course** [comment]
> Adobe Audition @Engineering-Course.

Thanks~

**Comment by dnnnew** [comment]
> However, the wav files on my dataset have the complete sentence and no silence.
> Does sampling_rate matter? @rafaelvalle
> The sampling rate of my wav is 48000. Then I change the sampling rate in hparams.py from 22050 to 48000.

I trained LJ-Speech dataset (same hparams which I trained my dataset) and it looked no gaps.

Since the wav files on your dataset have the complete sentence and no silence, maybe should we change attention network size?

**Comment by Engineering-Course** [comment]
I also have trained LJ-Speech dataset and it sounds great. And I only change the sampling rate in hparams to train my dataset.
What's your suggestion for attention network size?

**Comment by dnnnew** [comment]
>I also have trained LJ-Speech dataset and it sounds great. And I only change the sampling rate in hparams to train my dataset.

I changed these 
filter_length=2048,
hop_length=275,
win_length=1100 then my dataset learn attention (get aligned but with gaps).

> What's your suggestion for attention network size?

Before You told me about that the wav files on your dataset have the complete sentence and no silence, I never thought about it.

I'll try it later.  (maybe increase)

**Comment by Engineering-Course** [comment]
Thanks. I'll try it.

**Comment by tsungruihon** [comment]
may i ask how to generate alignment image? i use `tensorborad --logdir=output/log_dir ` but it just not show me the `alignment image`

**Comment by delgerdalai** [comment]
@tsungruihon 

https://github.com/NVIDIA/tacotron2/blob/f2c94d94fd64041bd8fdbd908beceea35e82ace0/hparams.py#L13

Alignment image is generated every 1000th iteration in the TensorBoard -> Images. 

![screen shot 2019-02-21 at 10 58 41 pm](https://user-images.githubusercontent.com/3131525/53178549-019ceb00-362d-11e9-838f-642c38423ec6.png)

**Comment by rafaelvalle** [comment]
@dnnnew did you try training with the default params in the repo?

**Comment by dnnnew** [comment]
@rafaelvalle, I did. But after 200k iters (415 epochs) with default params, it didn't learn attention. 
I changed params like above (from https://github.com/Rayhane-mamah/Tacotron-2), then it looks like this image just in 16k iters.

<img width="646" alt="screen shot 2019-02-22 at 10 51 00 am" src="https://user-images.githubusercontent.com/45612997/53216530-d8fa0d00-368f-11e9-8ccb-13d640dfa96f.png">

**Comment by rafaelvalle** [comment]
Please share the exact hparams you used to reach alignment in 16k iters.

**Comment by dnnnew** [comment]
I changed 

_hparams.py_
<img width="701" alt="screen shot 2019-02-22 at 11 08 22 am" src="https://user-images.githubusercontent.com/45612997/53217229-3beca380-3692-11e9-9bb9-2104a195e542.png">
batch_size=36

_symbols.py_
_letters = 'АБВГДЕЁЖЗИЙКЛМНОӨПРСТУҮФХЦЧШЪЫЬЭЮЯабвгдеёжзийклмноөпрстуүфхцчшъыьэюя'

I trained 2 GPU.

**Comment by dnnnew** [comment]
Also I changed only 

_filter_length=2048_

But it didn't learn attention.

**Comment by tsungruihon** [comment]
@dnnnew may i ask if the `train.csv` has numbers, like `123456`, does the `_characters` in `symbols.py` should contain these six numbers?

**Comment by delgerdalai** [comment]
You have to convert number to words. 
1 => one
2 => two 
etc .... 
using characters in symbols.py

or 

you can write custom cleaner function like that  https://github.com/NVIDIA/tacotron2/blob/master/text/cleaners.py

**Comment by rafaelvalle** [comment]
@dnnnew for sanity checking: when you ran for 200k and it did not learn attention, did you use the cyrillic symbols?

**Comment by dnnnew** [comment]
@rafaelvalle 
>when you ran for 200k and it did not learn attention, did you use the cyrillic symbols?

Yes

>i notice you're loading mels from disk. did you use the same mels to train the model from both repos?

No, 
I use preprocess.py and it generate mels with current hparams.
I got mels generation codes from here https://github.com/NVIDIA/tacotron2/blob/f2c94d94fd64041bd8fdbd908beceea35e82ace0/data_utils.py#L37. 
<img width="879" alt="screen shot 2019-02-22 at 4 27 24 pm" src="https://user-images.githubusercontent.com/45612997/53229741-052d8200-36c0-11e9-98bc-5d15aaca57b1.png">

I just wanted to reduce CPU load.

**Comment by tsungruihon** [comment]
@dnnnew May i ask you for some idea about my `alignment` image ? Or how could i judge the effectiveness by the `alignment` image that i have ?
<img width="669" alt="screen shot 2019-02-22 at 10 53 21 pm" src="https://user-images.githubusercontent.com/11624203/53250126-be0db400-36f4-11e9-80d3-700a048e55cf.png">

**Comment by rafaelvalle** [comment]
@dnnnew did you try training taco2 from this repo with mels produced from Rayhane-mamah repo?

**Comment by tsungruihon** [comment]
@rafaelvalle @dnnnew  Cause i use chinese corpus,  i changed the `hparams.py` into below:
```
 hparams = tf.contrib.training.HParams(
        ################################
        # Experiment Parameters        #
        ################################
        epochs=500,
        iters_per_checkpoint=1000,
        seed=1234,
        dynamic_loss_scaling=True,
        fp16_run=False,
        distributed_run=False,
        dist_backend="nccl",
        dist_url="tcp://localhost:54321",
        cudnn_enabled=True,
        cudnn_benchmark=False,

        ################################
        # Data Parameters             #
        ################################
        load_mel_from_disk=False,
        training_files='filelists/amber/amber_audio_text_train_filelist.txt',
        validation_files='filelists/amber/amber_audio_text_val_filelist.txt',
#        text_cleaners=['english_cleaners'],
        text_cleaners = ['basic_cleaners'],
        ################################
        # Audio Parameters             #
        ################################
        max_wav_value=32768.0,
        sampling_rate=48000,
        filter_length=4096,
        hop_length=600,
        win_length=2400,
        n_mel_channels=160,
        mel_fmin=125,
        mel_fmax=7600 
```

And in the `7000 steps`, the `alignment` image like this below:

<img width="680" alt="screen shot 2019-02-23 at 8 50 03 am" src="https://user-images.githubusercontent.com/11624203/53279054-6731ca00-3748-11e9-9117-a38f0924fe50.png">

Does it make sense? Or i should need to modify `hparams.py` again ?

**Comment by dnnnew** [comment]
> @dnnnew did you try training taco2 from this repo with mels produced from Rayhane-mamah repo?

@rafaelvalle No. I didn't.

**Comment by rafaelvalle** [comment]
@dnnnew give it a try.

**Comment by dnnnew** [comment]
@rafaelvalle Ok, I'll try it. Thanks.

**Comment by tugstugi** [comment]
@dnnnew @delgerdalai 
You can still keep the default parameter of this repo and train a Mongolian dataset. You have to cheat a little bit: load the pretrained LJSpeech model but without the embedding weight (because the Mongolian and English symbols differ). With a 5h Mongolian Bible dataset, I got already alignment starting from the step 2500 with the Noam LR scheduling.

**Comment by dnnnew** [comment]
@tugstugi 

Super cool :). I'm really glad to hear that. 
Here is a alignment plot of MBS dataset with default params (26k iters). 
It seems becoming with the gaps, so I stop training. 

<img width="688" alt="screen shot 2019-02-25 at 9 00 59 am" src="https://user-images.githubusercontent.com/45612997/53308513-e952f780-38dc-11e9-8b3f-a396aad5e298.png">
 
Does your alignment plot has a little gaps or not?

**Comment by tugstugi** [comment]
@dnnnew this repo uses scipy.io.wavfile.read and assumes that WAV files are int16 which is not good :) The WAV files of the Mongolian Bible should be float32. Have you fixed that before training the Mongolian Bible dataset?

So far I remember, the alignments looked OK. I can check tomorrow my dev machine whether the plots looked OK or not.

**Comment by dnnnew** [comment]
@tugstugi 

First, I disabled this line.
https://github.com/NVIDIA/tacotron2/blob/f2c94d94fd64041bd8fdbd908beceea35e82ace0/data_utils.py#L43
But I get error some files of MBS. 

Then, I saved as MBS dataset to int16 😥, and didn't change code.

Thank you

**Comment by tugstugi** [comment]
@dnnnew somehow I removed the logs. Found only an alignment image of the 3750th step:
![3750](https://user-images.githubusercontent.com/243826/53326400-c7aa3e00-38e5-11e9-87f8-0a17a69eaf99.png)

**Comment by dnnnew** [comment]
Nice, it looks no gaps. 

I've started training with pertained LJ weights but without Noam LR scheduler. 
How important Noam LR scheduler?

Thanks again.

**Comment by rafaelvalle** [comment]
@tugstugi thanks for sharing these. With the current setup on LJS there are no gaps in the alignments. For this reason, I suspect that the gaps you are seeing are related to your text and data pre-processing.

**Comment by tugstugi** [comment]
@dnnnew  I prefer the Noam LR scheduling specially if I am using a pretrained model. It converges much faster than with a fixed big LR. From the 5000th step, the generated Mongolian audio using Griffin-Lim should be already intelligable.

**Comment by dnnnew** [comment]
@tugstugi , @rafaelvalle 
LJS pretrained weights and after 8k iters. (default params)
<img width="661" alt="screen shot 2019-02-26 at 8 35 16 am" src="https://user-images.githubusercontent.com/45612997/53378686-7c556580-39a1-11e9-9887-6363ed6f98de.png">

[LJS-pretrained_8k_step.wav.zip](https://github.com/NVIDIA/tacotron2/files/2903225/LJS-pretrained_8k_step.wav.zip)

Without pertained weights and after 202k iters. (audio params was changed)
<img width="671" alt="screen shot 2019-02-26 at 8 27 02 am" src="https://user-images.githubusercontent.com/45612997/53378758-c1799780-39a1-11e9-874d-583180ec7b3b.png">
[202k_step_without_pretrain.wav.zip](https://github.com/NVIDIA/tacotron2/files/2903227/202k_step_without_pretrain.wav.zip)

Sound files are generated by Waveglow(500k , 200 epochs, "filter_length": 2048,"hop_length": 275,"win_length": 1100, n_channels: 256).

Thank you all.

**Comment by tugstugi** [comment]
@dnnnew I have checked your attached training sample. Actually your speaker makes many unnatural pauses after every few words like "гэх мэдээлэл", "хэвлэлээр цацагдсан нь" and "ор үндэсгүй гэж". 

Who is the speaker? The voice sounds familiar.

**Comment by dnnnew** [comment]
Yep :(. 
Also, we made a mistake when we prepare dataset.  Our training sample has multiple sentence, or incomplete sentences. We are working on realignment sound and text. 

Нарамчимэг гэсэн байхаа, үндэсний радиод ажиллаж байсан нэг эгчээр уншуулсан. Бид нар мэдээ уншдаг модел-тэй болох гээд нээлттэй кодуудыг туршиж үзээд.

**Comment by tugstugi** [comment]
@dnnnew Is your waveglow trained on your dataset? Could you try something for me? Feed your Mongolian Tacotron output to the LJSpeech trained waveglow and post the result here :)

PS: send me message over FB. I have some private questions.

**Comment by dnnnew** [comment]
Yes, waveglow is trained on my dataset.

Tacotron model (LJS pertained weights + 23k iters /training is ongoing/) + LJS pretrained waveglow / my waveglow(500k iters). 

Here is result. 😮 super
[result.zip](https://github.com/NVIDIA/tacotron2/files/2906206/result.zip)


PS: Sure

**Comment by tugstugi** [comment]
so LJSpeech waveglow works better than Mongolian trained waveglow for Mongolian texts? :D

**Comment by dnnnew** [comment]
I read that LJS waveglow is trained 870epochs, my version is trained 130 epochs.
Is it possible fine tuning LJS waveglow with my dataset?

**Comment by tugstugi** [comment]
@dnnnew sure, you can finetune but the problem is waveglow's train code and their provided pretrained model don't match. So you have to fix it first.

**Comment by yondu22** [comment]
what does this plot mean, can someone explain it?
